admin/test/test_release.py

```diff
@@ -688,6 +688,138 @@ def test_publish_non_release_fails(self):
             aws, '0.3.0-444-gf05215b', '0.3.0-444-gf05215b',
             environment=Environments.STAGING)
 
+    def assert_error_key_update(self, doc_version, environment, should_update):
+        """
+        Call ``publish_docs`` and assert that only the expected buckets have an
+        updated error_key property.
+
+        :param unicode doc_version: The version of the documentation that is
+            being published.
+        :param NamedConstant environment: One of the ``NamedConstants`` in
+            ``Environments``.
+        :param bool should_update: A flag indicating whether the error_key for
+            the bucket associated with ``environment`` is expected to be
+            updated.
+        :raises: ``FailTest`` if an error_key in any of the S3 buckets has been
+            updated unexpectedly.
+        """
+        # Get a set of all known S3 buckets.
+        bucket_names = set()
+        for e in Environments.iterconstants():
+            bucket_names.add(
+                DOCUMENTATION_CONFIGURATIONS[e].documentation_bucket
+            )
+            bucket_names.add(
+                DOCUMENTATION_CONFIGURATIONS[e].dev_bucket
+            )
+        # Pretend that both devel and latest aliases are currently pointing to
+        # an older version.
+        empty_routes = {
+            'en/devel/': 'en/0.0.0/',
+            'en/latest/': 'en/0.0.0/',
+        }
+        # In all the S3 buckets.
+        empty_routing_rules = {bucket_name: empty_routes.copy() for bucket_name in bucket_names}
+        # And that all the buckets themselves are empty.
+        empty_buckets = {bucket_name: {} for bucket_name in bucket_names}
+        # And that all the buckets have an empty error_key
+        empty_error_keys = {bucket_name: {} for bucket_name in bucket_names}
```

Comment:
An empty `dict` isn't the right type for a default value. In any case, the dev bucket probably shouldn't have an error key set.